internal/caller/linux/linuxconfig.go

```diff
@@ -0,0 +1,72 @@
+package caller
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+const (
+	installPath = "/etc/wireguard/"
+)
+
+// LinuxConfig represents the tunnel configuration for a linux machine.
+type LinuxConfig struct {
+	DeviceName string
+	Config     []byte
+}
+
+// NewConfig generates and returns a new LinuxConfig.
+func NewConfig() *LinuxConfig {
+	return new(LinuxConfig)
+}
+
+// InstallConfig installs the wireguard tunnel to the machine.
+func (l *LinuxConfig) InstallConfig() error {
+	filename := installPath + l.DeviceName
+
+	// Check the existence of the wireguard directory
+	if _, err := os.Stat(installPath); os.IsNotExist(err) {
+		os.Mkdir(installPath, 0700)
+	}
+
+	// Check the existence of a wireguard conf
+	if _, err := os.Stat(filename); os.IsExist(err) {
+		if e := os.Remove(filename); e != nil {
+			return e
+		}
+	}
+
+	// Write file
+	if err := ioutil.WriteFile(filename, l.Config, 0644); err != nil {
```

Comment:
This file has private keys in it.  World readable is probably not the best choice here.  I suggest either 0640 or 0600.